In [1]:
import feather
import numpy as np
import pandas as pd
import os
from pathlib import Path

Feather data format is a lightweight as well as very fast binary format for storing data frames. Feather file, takes less than half of the space, than the corresponding CSV file, having same data. Feather files are 100 times faster, while reading and writing from the disk, as compared to CSV file

In [2]:
data_dir ='../input/tabular-playground-series-oct-2022/'
display(os.listdir(data_dir))

['sample_submission.csv',
 'train_9.csv',
 'train_3.csv',
 'train_4.csv',
 'test_dtypes.csv',
 'train_8.csv',
 'train_dtypes.csv',
 'train_5.csv',
 'train_2.csv',
 'test.csv',
 'train_0.csv',
 'train_7.csv',
 'train_1.csv',
 'train_6.csv']

# Convert data into feather

In [3]:

%%time
n=10
for i in range(n):
    data = pd.read_csv(data_dir + f'/train_{i}.csv')
    data.to_feather(f'train_{i}.feather')
    print('Complete Transfer on Feather :', i)
test =pd.read_csv(data_dir + "test.csv")
test.to_feather('test.feather')
print("Done")

Complete Transfer on Feather : 0
Complete Transfer on Feather : 1
Complete Transfer on Feather : 2
Complete Transfer on Feather : 3
Complete Transfer on Feather : 4
Complete Transfer on Feather : 5
Complete Transfer on Feather : 6
Complete Transfer on Feather : 7
Complete Transfer on Feather : 8
Complete Transfer on Feather : 9
Done
CPU times: user 7min 53s, sys: 35.6 s, total: 8min 29s
Wall time: 9min 29s


# Check SpeeD

In [7]:
outdir = Path('.')
os.makedirs(str(outdir), exist_ok=True)

CPU times: user 326 µs, sys: 27 µs, total: 353 µs
Wall time: 376 µs


In [8]:
%%time
#load with feather data
train_0 = pd.read_feather(outdir / 'train_0.feather')
train_0


CPU times: user 1.77 s, sys: 2.18 s, total: 3.95 s
Wall time: 1.39 s


game_num  event_id  event_time  ball_pos_x  ball_pos_y  ball_pos_z  \
0               1      1002  -33.313030     -0.0000    0.000000    1.854800   
1               1      1002  -33.206146     -0.0000    0.000000    1.854800   
2               1      1002  -33.098114     -0.0000    0.000000    1.854800   
3               1      1002  -32.993190     -0.0000    0.000000    1.854800   
4               1      1002  -32.887756     -0.0000    0.000000    1.854800   
...           ...       ...         ...         ...         ...         ...   
2149376       737    737005   -0.473386     11.6650   25.500801   15.000999   
2149377       737    737005   -0.363688     10.3796   31.147000   18.421200   
2149378       737    737005   -0.254574      9.1902   36.371002   21.428001   
2149379       737    737005   -0.145479      8.1868   40.778202   23.845400   
2149380       737    737005   -0.036366      7.0956   45.572200   26.350200   

         ball_vel_x  ball_vel_y  ball_vel_z  p0_pos_x  ...  boost0_timer  \
0           -0.0000    0.000000      0.0000   41.8048  ...           0.0   
1           -0.0000    0.000000      0.0000   42.2616  ...           0.0   
2           -0.0000    0.000000      0.0000   43.2270  ...           0.0   
3           -0.0000    0.000000      0.0000   43.8984  ...           0.0   
4           -0.0000    0.000000      0.0000   44.9606  ...           0.0   
...             ...         ...         ...       ...  ...           ...   
2149376    -11.0376   48.485400     30.1854   36.1764  ...           0.0   
2149377    -10.9984   48.311802     28.5632   36.6466  ...           0.0   
2149378    -10.9620   48.150600     27.0618   37.1824  ...           0.0   
2149379    -10.9312   48.016200     25.7954   37.6878  ...           0.0   
2149380    -10.8976   47.869800     24.4178   38.4584  ...           0.0   

         boost1_timer  boost2_timer  boost3_timer  boost4_timer  boost5_timer  \
0               0.000         0.000         0.000          0.00         0.000   
1               0.000         0.000         0.000          0.00         0.000   
2               0.000         0.000         0.000          0.00         0.000   
3               0.000         0.000         0.000          0.00         0.000   
4               0.000         0.000         0.000          0.00         0.000   
...               ...           ...           ...           ...           ...   
2149376        -6.918        -2.748        -2.418         -9.05        -4.574   
2149377        -6.810        -2.639        -2.310         -8.94        -4.465   
2149378        -6.700        -2.531        -2.201         -8.83        -4.360   
2149379        -6.590        -2.422        -2.092         -8.72        -4.250   
2149380        -6.484        -2.312        -1.983         -8.61        -4.140   

         player_scoring_next  team_scoring_next  team_A_scoring_within_10sec  \
0                          3                  B                            0   
1                          3                  B                            0   
2                          3                  B                            0   
3                          3                  B                            0   
4                          3                  B                            0   
...                      ...                ...                          ...   
2149376                   -1               None                            0   
2149377                   -1               None                            0   
2149378                   -1               None                            0   
2149379                   -1               None                            0   
2149380                   -1               None                            0   

         team_B_scoring_within_10sec  
0                                  0  
1                                  0  
2                                  0  
3                                  0  
4                 

Now, Lets read the csv files and see the difference between. feather data take 1.39 second wall time and what about CSV file? 54.1 second, can you Imagine?

In [10]:
%%time
#load csv data
train_0 = pd.read_csv(data_dir + 'train_0.csv')
train_0

CPU times: user 44.4 s, sys: 4.08 s, total: 48.5 s
Wall time: 54.1 s


game_num  event_id  event_time  ball_pos_x  ball_pos_y  ball_pos_z  \
0               1      1002  -33.313030     -0.0000    0.000000    1.854800   
1               1      1002  -33.206146     -0.0000    0.000000    1.854800   
2               1      1002  -33.098114     -0.0000    0.000000    1.854800   
3               1      1002  -32.993190     -0.0000    0.000000    1.854800   
4               1      1002  -32.887756     -0.0000    0.000000    1.854800   
...           ...       ...         ...         ...         ...         ...   
2149376       737    737005   -0.473386     11.6650   25.500801   15.000999   
2149377       737    737005   -0.363688     10.3796   31.147000   18.421200   
2149378       737    737005   -0.254574      9.1902   36.371002   21.428001   
2149379       737    737005   -0.145479      8.1868   40.778202   23.845400   
2149380       737    737005   -0.036366      7.0956   45.572200   26.350200   

         ball_vel_x  ball_vel_y  ball_vel_z  p0_pos_x  ...  boost0_timer  \
0           -0.0000    0.000000      0.0000   41.8048  ...           0.0   
1           -0.0000    0.000000      0.0000   42.2616  ...           0.0   
2           -0.0000    0.000000      0.0000   43.2270  ...           0.0   
3           -0.0000    0.000000      0.0000   43.8984  ...           0.0   
4           -0.0000    0.000000      0.0000   44.9606  ...           0.0   
...             ...         ...         ...       ...  ...           ...   
2149376    -11.0376   48.485400     30.1854   36.1764  ...           0.0   
2149377    -10.9984   48.311802     28.5632   36.6466  ...           0.0   
2149378    -10.9620   48.150600     27.0618   37.1824  ...           0.0   
2149379    -10.9312   48.016200     25.7954   37.6878  ...           0.0   
2149380    -10.8976   47.869800     24.4178   38.4584  ...           0.0   

         boost1_timer  boost2_timer  boost3_timer  boost4_timer  boost5_timer  \
0               0.000         0.000         0.000          0.00         0.000   
1               0.000         0.000         0.000          0.00         0.000   
2               0.000         0.000         0.000          0.00         0.000   
3               0.000         0.000         0.000          0.00         0.000   
4               0.000         0.000         0.000          0.00         0.000   
...               ...           ...           ...           ...           ...   
2149376        -6.918        -2.748        -2.418         -9.05        -4.574   
2149377        -6.810        -2.639        -2.310         -8.94        -4.465   
2149378        -6.700        -2.531        -2.201         -8.83        -4.360   
2149379        -6.590        -2.422        -2.092         -8.72        -4.250   
2149380        -6.484        -2.312        -1.983         -8.61        -4.140   

         player_scoring_next  team_scoring_next  team_A_scoring_within_10sec  \
0                          3                  B                            0   
1                          3                  B                            0   
2                          3                  B                            0   
3                          3                  B                            0   
4                          3                  B                            0   
...                      ...                ...                          ...   
2149376                   -1                NaN                            0   
2149377                   -1                NaN                            0   
2149378                   -1                NaN                            0   
2149379                   -1                NaN                            0   
2149380                   -1                NaN                            0   

         team_B_scoring_within_10sec  
0                                  0  
1                                  0  
2                                  0  
3                                  0  
4                 


I save the dataset to <a href="https://www.kaggle.com/datasets/gazu468/tpsoct22-feather-files"> click here</a>.Thanks for read this notebook.